tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,484 @@
+                # mysql has no exporter defined, so everything is via Pandas
+                addl_hints = pandas_compatible_addl_hints
+            elif self.engine.name == 'postgresql':
+                if dateformat == 'YYYY-MM-DD':
+                    # ensure we keep other areas of ISO style
+                    addl_hints = {
+                        'timeonlyformat': 'HH:MI:SS',
+                        'datetimeformattz': 'YYYY-MM-DD HH:MI:SS',
+                        'datetimeformat': 'YYYY-MM-DD HH:MI:SS',
+                    }
+                else:
+                    # We will be using pandas
+                    addl_hints = pandas_compatible_addl_hints
+            elif self.engine.name == 'vertica':
+                # Make sure our '\n' strings below are valid when comparing output
+                addl_hints['record-terminator'] = '\n'
+                if dateformat != 'YYYY-MM-DD':
+                    # We will be using pandas
+                    addl_hints.update(pandas_compatible_addl_hints)
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'dateformat': dateformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            csv_text = self.unload(column_name='date',
+                                   records_format=records_format)
+            self.assertEqual(csv_text, create_sample(dateformat) + "\n",
+                             f"from dateformat {dateformat} and addl_hints {addl_hints}")
+
+    def test_unload_datetime(self) -> None:
+        self.createDateTimeTable()
+        matching_dateformat = {
+            'YYYY-MM-DD HH:MI:SS': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH12:MI AM': 'YYYY-MM-DD',
+            'MM/DD/YY HH24:MI': 'MM/DD/YY',
+            'YYYY-MM-DD HH24:MI:SS': 'YYYY-MM-DD',
+        }
+        for datetimeformat in DATETIME_CASES:
+            addl_hints: PartialRecordsHints = {}
+            pandas_compatible_addl_hints: PartialRecordsHints = {
+                'dateformat': matching_dateformat[datetimeformat],
+                'datetimeformattz': datetimeformat,
+            }
+            if self.engine.name == 'redshift':
+                if datetimeformat != 'YYYY-MM-DD HH24:MI:SS':
+                    # this is the only format supported by Redshift on
+                    # export, so we're going to need to be sure to use
+                    # hints that work in Pandas - i.e.,
+                    addl_hints = pandas_compatible_addl_hints
+                    if 'AM' in datetimeformat:
+                        # TODO: Add a GitHub issue for this
+                        # TODO: Should expect failure and complain if it doesn't fail
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker